### PR TITLE
Automated synchronizing DRPM package via API/CLI/UI

### LINKED ISSUE #3852: Automate synchronizing DRPM package via API/CLI/UI
# Request

Since **pulp** allows one to synchronize `delta rpms` (aka `drpms`), we need to make sure that we can perform the synchronization of repositories containing this type of content.

## Setup

* Create a new organization **ORG**
* Create a new lifecycle environment **LCE** with **LIBRARY** as its prior environment
* Create a new product **PROD**
* Create a new cusotm repository **REPO** with a feed pointed at ~~`https://repos.fedorapeople.org/repos/pulp/pulp/demo_repos/test_drpm_repo/`~~  `https://repos.fedorapeople.org/repos/pulp/pulp/fixtures/drpm`

## Tests

* Synchronize content and assert that it is fully synchronized and that `drpms` can be listed
* Synchronize content, add product to a content view, publish it and make sure that `drpms` can be listed under the proper content view
* Synchronize content, add product to a content view, publish it, promote it to  and make sure that `drpms` can be listed under the proper content view

## Validation

Since `drpms` are **not** displayed on Satellite interface, there are three possible ways to validate that this type of content has been successfully synchronized:

* Look under `/var/lib/pulp/published/yum/https/repos/ORG/Library/custom/PROD/REPO/drpms/` and make sure that you see files with names ending in `.drpm`
* Open a connection (GET) against `http://SATELLITE_FQDN/pulp/repos/ORG/Library/custom/PROD/REPO/drpms/` and check that the listing returned contains packages with names ending in `.drpm`
* Create a new host and associate it with a product containing `drpm` packages. Run `yum install` or `dnf install` and check that the output generated shows the usage of `drpms`

I recommend we use either the first or second method.

### PR DESCRIPTION
**Pull request by abalakh**
Closes #3852 

Test results:
```python
py.test tests/foreman/{api,cli}/test_repository.py::DRPMRepositoryTestCase -v -n 2
================================================ test session starts ================================================
platform linux2 -- Python 2.7.10, pytest-2.9.2, py-1.4.31, pluggy-0.3.1 -- /home/andrii/env/bin/python
cachedir: .cache
rootdir: /home/andrii/workspace/robottelo, inifile: 
plugins: xdist-1.13.1, repeat-0.2
[gw0] linux2 Python 2.7.10 cwd: /home/andrii/workspace/robottelo
[gw1] linux2 Python 2.7.10 cwd: /home/andrii/workspace/robottelo
[gw0] Python 2.7.10 (default, Sep  8 2015, 17:20:17)  -- [GCC 5.1.1 20150618 (Red Hat 5.1.1-4)]
[gw1] Python 2.7.10 (default, Sep  8 2015, 17:20:17)  -- [GCC 5.1.1 20150618 (Red Hat 5.1.1-4)]
gw0 [6] / gw1 [6]
scheduling tests via LoadScheduling

tests/foreman/api/test_repository.py::DRPMRepositoryTestCase::test_positive_sync_publish_promote_cv 
tests/foreman/api/test_repository.py::DRPMRepositoryTestCase::test_positive_sync 
[gw0] PASSED tests/foreman/api/test_repository.py::DRPMRepositoryTestCase::test_positive_sync 
tests/foreman/api/test_repository.py::DRPMRepositoryTestCase::test_positive_sync_publish_cv 
[gw0] PASSED tests/foreman/api/test_repository.py::DRPMRepositoryTestCase::test_positive_sync_publish_cv 
tests/foreman/cli/test_repository.py::DRPMRepositoryTestCase::test_positive_sync_publish_cv 
[gw1] PASSED tests/foreman/api/test_repository.py::DRPMRepositoryTestCase::test_positive_sync_publish_promote_cv 
[gw0] PASSED tests/foreman/cli/test_repository.py::DRPMRepositoryTestCase::test_positive_sync_publish_cv 
tests/foreman/cli/test_repository.py::DRPMRepositoryTestCase::test_positive_sync_publish_promote_cv 
tests/foreman/cli/test_repository.py::DRPMRepositoryTestCase::test_positive_sync 
[gw1] PASSED tests/foreman/cli/test_repository.py::DRPMRepositoryTestCase::test_positive_sync 
[gw0] PASSED tests/foreman/cli/test_repository.py::DRPMRepositoryTestCase::test_positive_sync_publish_promote_cv 

============================================ 6 passed in 244.08 seconds =============================================
py.test tests/foreman/ui/test_repository.py -v -n 2 -k 'test_positive_drpm'
================================================ test session starts ================================================
platform linux2 -- Python 2.7.10, pytest-2.9.2, py-1.4.31, pluggy-0.3.1 -- /home/andrii/env/bin/python
cachedir: .cache
rootdir: /home/andrii/workspace/robottelo, inifile: 
plugins: xdist-1.13.1, repeat-0.2
[gw0] linux2 Python 2.7.10 cwd: /home/andrii/workspace/robottelo
[gw1] linux2 Python 2.7.10 cwd: /home/andrii/workspace/robottelo
[gw0] Python 2.7.10 (default, Sep  8 2015, 17:20:17)  -- [GCC 5.1.1 20150618 (Red Hat 5.1.1-4)]
[gw1] Python 2.7.10 (default, Sep  8 2015, 17:20:17)  -- [GCC 5.1.1 20150618 (Red Hat 5.1.1-4)]
gw0 [3] / gw1 [3]
scheduling tests via LoadScheduling

tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_drpm_sync 
[gw0] PASSED tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_drpm_sync 
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_drpm_sync_publish_cv 
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_drpm_sync_publish_promote_cv 
[gw0] PASSED tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_drpm_sync_publish_cv 
[gw1] PASSED tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_drpm_sync_publish_promote_cv 

============================================ 3 passed in 236.30 seconds =============================================
```

### CODE REVIEW

**Comment by abalakh** [comment]
I was thinking about combining these tests with SRPM sync tests, as the tests are pretty similar so we could use subtests. But, the difference is in repo we use, path to folder and file extension. So it will end in some overwhelmed commands like:
```python
        result = ssh.command(
            'ls /var/lib/pulp/published/yum/https/repos/{}/{}/{}/custom/{}/{}'
            '/{}/ | grep {}'
            .format(
                self.org.label,
                lce.name,
                cv.label,
                self.product.label,
                repo.label,
                repo_type['subdir'],
                repo_type['extension'],
            )
        )
```
Also it will become harder to maintain such test and, as @omaciel pointed, it won't be so obvious which feature is broken by simply looking through jenkins results.
In other words - these tests can be combined, but i can't see much profit in it. Feel free to correct me if subtests would be better option in your opinion (the only advantage i see so far is reducing duplicated code).

**Comment by sghai** [comment]
ACK.. Looks good and I prefer to have them separated..

**Comment by sghai** [comment]
@SatelliteQE/quality-engineers : one more reviewer please.

**Comment by omaciel** [comment]
@abalakh 

```console
2016-10-04 18:00:24 < omaciel> abalakht++
2016-10-04 18:00:25 < robottelo> abalakht has 21 points
```
